subgraphs/compound-v3/src/sdk/account.ts

```diff
@@ -0,0 +1,331 @@
+import {
+  BigDecimal,
+  BigInt,
+  Bytes,
+  ethereum,
+  log,
+} from "@graphprotocol/graph-ts";
+import {
+  Account,
+  LendingProtocol,
+  Market,
+  Position,
+  PositionSnapshot,
+  PositionCounter,
+} from "../../generated/schema";
+import {
+  BIGINT_ZERO,
+  exponentToBigDecimal,
+  INT_ONE,
+  INT_ZERO,
+  TransactionType,
+} from "./constants";
+import { TokenManager } from "./token";
+
+/**
+ * This file contains the AccountClass, which does
+ * the operations on the Account entity. This includes:
+ *  - Creating a new Account
+ *  - Updating an existing Account
+ *  - Making a position
+ *  - Making position snapshots
+ *
+ * Schema Version: 3.0.0
+ * Last Updated: Dec 4, 2022
+ * Author(s):
+ *  - @dmelotik
+ */
+export class AccountManager {
+  private isNew: boolean; // true if the account was created
+  private account: Account;
+  private positonError: boolean;
+  private position!: Position;
+  private market: Market;
+  private protocol: LendingProtocol;
+  private event: ethereum.Event;
+
+  constructor(
+    account: Bytes,
+    market: Market,
+    protocol: LendingProtocol,
+    event: ethereum.Event
```

Comment:
You should separate concerns here. The existence of an account doesn't need to be attached to some market. The actions it might perform are, but you should be able to instantiate accounts on their own. It might be the case that all supported account actions are always related to a Market now, but if we eventually added other non-market entities with which an account can interact (say some token staking, governance stuff, etc.) this would need to be decoupled.

Instead, when calling a method where a market is necessary, pass it as an argument.